lib/internal/main/worker_thread.js

```diff
@@ -163,6 +140,68 @@ process._fatalException = (error) => {
   }
 };
 
+function makeEnvProxy(env) {
+  let ret;
+
+  const handler = {
+    set(target, key, value) {
+      // Override 'set' to make sure only strings are stored in our copy of
+      // `process.env`. The deprecation warning text and behaviour matches
+      // what the main thread does in C++.
+      if (typeof value !== 'string' &&
+          typeof value !== 'number' &&
+          typeof value !== 'boolean') {
+        const { getOptionValue } = require('internal/options');
+        if (getOptionValue('--pending-deprecation')) {
+          process.emitWarning(
+            'Assigning any value other than a string, number, or boolean to ' +
+            'a process.env property is deprecated. Please make sure to ' +
+            'convert the value to a string before setting process.env with it.',
+            'DeprecationWarning', 'DEP0104');
+        }
+      }
+      return Reflect.set(target, `${key}`, `${value}`);
+    },
+
+    defineProperty(target, key, descriptor) {
+      return handler.set(
+        target, key, typeof descriptor.get === 'undefined' ?
+          descriptor.value : descriptor.get.call(ret));
+    }
+  };
+
+  ret = new Proxy(env, handler);
+  return ret;
+}
+
+function workerSpecificSetup() {
+  // Since worker threads cannot switch cwd, we do not need to
+  // overwrite the process.env.NODE_V8_COVERAGE variable.
+  if (process.env.NODE_V8_COVERAGE) {
```

Comment:
I think these can be done as soon as possible, and not in the `LOAD_SCRIPT` message handler? The same goes to `makeEnvProxy` - the values of process.env will be inconsistent for internal modules loaded that access process.env before the `LOAD_SCRIPT` message is handled.

Reply:
That means we have to transfer `process.env` in another way, which also adds a ton of extra complexity…

Reply:
Then maybe we should just remove the creation of process.env in `RunBootstrapping`, and only do it for the main thread? Then until `makeEnvProxy()` is called `process.env` will be undefined. Does any internal module currently access `process.env` before the `LOAD_SCRIPT` message is handled?